File: src/main/java/fi/quanfoxes/parser/nodes/NegateNode.java
package fi.quanfoxes.parser.nodes;

import fi.quanfoxes.parser.Node;

public class NegateNode extends Node {
    public NegateNode(Node object) {
        super.add(object);
    }

    @Override
    public NodeType getNodeType() {
        return NodeType.NEGATE_NODE;
    }
}

File: Zigzag/Parser/Nodes/NegateNode.cs
public class NegateNode : Node, IType
{
	public Node Target => First!;

	public NegateNode(Node target)
	{
		Add(target);
	}

	public new Type? GetType()
	{
		return Target is IType x ? x.GetType() : Types.UNKNOWN;
	}

	public override NodeType GetNodeType()
	{
		return NodeType.NEGATE_NODE;
	}
}